ui/src/app/views/dataset/component.tsx

```diff
@@ -58,15 +92,102 @@ export const DataSet = withCurrentWorkspace()(class extends React.Component<
       const {namespace, id} = this.props.workspace;
       const conceptSets = await Promise.resolve(conceptSetsApi()
         .getConceptSetsInWorkspace(namespace, id));
-      this.setState({conceptSetList: conceptSets.items});
+      this.setState({conceptSetList: conceptSets.items, loadingConceptSets: false});
     } catch (error) {
       console.log(error);
     }
   }
 
+  convertResource(r: ConceptSet | Cohort, rType: ResourceType): RecentResource {
+    const {workspace} = this.props;
+    this.setState({rType: rType});
+    console.log(rType);
+    return convertToResource(r, workspace.namespace, workspace.id,
+      workspace.accessLevel as unknown as WorkspaceAccessLevel, rType);
+  }
+
+  openConfirmDelete(r: ConceptSet, rType: ResourceType): void {
+    const rc = this.convertResource(r, rType);
+    this.setState({confirmDeleting: true, resource: rc});
+  }
+
+  closeConfirmDelete(): void {
```

Comment:
Technically it should only do this if you actually click confirm, correct? Semantically this seems like it is called no matter how the confirm delete modal is closed. (Maybe: `onConfirmDelete`?)

Or at least, the filtering out of the list should only be on delete.

Reply:
This was one of the extra thoughts I had after putting this up last night. I will sort this logic appropriately. (just close if cancel hit, etc.)

Reply:
Fixed for both modals